monitoring/instance_manager_collector.go

```diff
@@ -0,0 +1,174 @@
+package monitoring
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metricsclientset "k8s.io/metrics/pkg/client/clientset/versioned"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	"github.com/longhorn/longhorn-manager/types"
+)
+
+type InstanceManagerCollector struct {
+	*baseCollector
+
+	kubeMetricsClient *metricsclientset.Clientset
+	namespace         string
+
+	instanceManagerUsageCPUMetric      metricInfo
+	instanceManagerUsageMemoryMetric   metricInfo
+	instanceManagerRequestCPUMetric    metricInfo
+	instanceManagerRequestMemoryMetric metricInfo
+}
+
+func NewInstanceManagerCollector(
+	logger logrus.FieldLogger,
+	nodeID string,
+	ds *datastore.DataStore,
+	kubeMetricsClient *metricsclientset.Clientset,
+	namespace string) *InstanceManagerCollector {
+
+	imc := &InstanceManagerCollector{
+		baseCollector:     newBaseCollector("instance-manager", logger, nodeID, ds),
+		kubeMetricsClient: kubeMetricsClient,
+		namespace:         namespace,
+	}
+
+	imc.instanceManagerUsageCPUMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemInstanceManager, "usage_cpu_millicpu"),
```

Comment:
Same, how about `cpu_usage_millicpu`?
Index by HW component then specific metric, same for the other fields below?

Reply:
If you look at the sentence below, my point might be more clear?
the `cpu usage` vs the `usage cpu / usage of cpu`

Reply:
Thank you! Fixed